{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyMFLEidPjQb3eN69AU67EH2",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Eitan177/cloneseer/blob/main/0904example_usage.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "cgRU4h5-gxmy",
        "outputId": "57f97a33-01cc-4430-9c79-ed1d283fefed"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-09-05 20:48:07--  https://ftp.ncbi.nih.gov/blast/executables/igblast/release/LATEST/ncbi-igblast-1.22.0-x64-linux.tar.gz\n",
            "Resolving ftp.ncbi.nih.gov (ftp.ncbi.nih.gov)... 130.14.250.12, 130.14.250.31, 130.14.250.7, ...\n",
            "Connecting to ftp.ncbi.nih.gov (ftp.ncbi.nih.gov)|130.14.250.12|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 46009645 (44M) [application/x-gzip]\n",
            "Saving to: ‘ncbi-igblast-1.22.0-x64-linux.tar.gz’\n",
            "\n",
            "ncbi-igblast-1.22.0 100%[===================>]  43.88M  69.6MB/s    in 0.6s    \n",
            "\n",
            "2025-09-05 20:48:08 (69.6 MB/s) - ‘ncbi-igblast-1.22.0-x64-linux.tar.gz’ saved [46009645/46009645]\n",
            "\n",
            "Collecting Bio\n",
            "  Downloading bio-1.8.0-py3-none-any.whl.metadata (5.7 kB)\n",
            "Collecting biopython>=1.80 (from Bio)\n",
            "  Downloading biopython-1.85-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (13 kB)\n",
            "Collecting gprofiler-official (from Bio)\n",
            "  Downloading gprofiler_official-1.0.0-py3-none-any.whl.metadata (11 kB)\n",
            "Collecting mygene (from Bio)\n",
            "  Downloading mygene-3.2.2-py2.py3-none-any.whl.metadata (10 kB)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.12/dist-packages (from Bio) (2.2.2)\n",
            "Requirement already satisfied: pooch in /usr/local/lib/python3.12/dist-packages (from Bio) (1.8.2)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.12/dist-packages (from Bio) (2.32.4)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.12/dist-packages (from Bio) (4.67.1)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.12/dist-packages (from biopython>=1.80->Bio) (2.0.2)\n",
            "Collecting biothings-client>=0.2.6 (from mygene->Bio)\n",
            "  Downloading biothings_client-0.4.1-py3-none-any.whl.metadata (10 kB)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.12/dist-packages (from pandas->Bio) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.12/dist-packages (from pandas->Bio) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.12/dist-packages (from pandas->Bio) (2025.2)\n",
            "Requirement already satisfied: platformdirs>=2.5.0 in /usr/local/lib/python3.12/dist-packages (from pooch->Bio) (4.4.0)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.12/dist-packages (from pooch->Bio) (25.0)\n",
            "Requirement already satisfied: charset_normalizer<4,>=2 in /usr/local/lib/python3.12/dist-packages (from requests->Bio) (3.4.3)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.12/dist-packages (from requests->Bio) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.12/dist-packages (from requests->Bio) (2.5.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.12/dist-packages (from requests->Bio) (2025.8.3)\n",
            "Requirement already satisfied: httpx>=0.22.0 in /usr/local/lib/python3.12/dist-packages (from biothings-client>=0.2.6->mygene->Bio) (0.28.1)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.12/dist-packages (from python-dateutil>=2.8.2->pandas->Bio) (1.17.0)\n",
            "Requirement already satisfied: anyio in /usr/local/lib/python3.12/dist-packages (from httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (4.10.0)\n",
            "Requirement already satisfied: httpcore==1.* in /usr/local/lib/python3.12/dist-packages (from httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (1.0.9)\n",
            "Requirement already satisfied: h11>=0.16 in /usr/local/lib/python3.12/dist-packages (from httpcore==1.*->httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (0.16.0)\n",
            "Requirement already satisfied: sniffio>=1.1 in /usr/local/lib/python3.12/dist-packages (from anyio->httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (1.3.1)\n",
            "Requirement already satisfied: typing_extensions>=4.5 in /usr/local/lib/python3.12/dist-packages (from anyio->httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (4.15.0)\n",
            "Downloading bio-1.8.0-py3-none-any.whl (321 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m321.1/321.1 kB\u001b[0m \u001b[31m7.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading biopython-1.85-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.3/3.3 MB\u001b[0m \u001b[31m62.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading gprofiler_official-1.0.0-py3-none-any.whl (9.3 kB)\n",
            "Downloading mygene-3.2.2-py2.py3-none-any.whl (5.4 kB)\n",
            "Downloading biothings_client-0.4.1-py3-none-any.whl (46 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m46.7/46.7 kB\u001b[0m \u001b[31m3.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: biopython, gprofiler-official, biothings-client, mygene, Bio\n",
            "Successfully installed Bio-1.8.0 biopython-1.85 biothings-client-0.4.1 gprofiler-official-1.0.0 mygene-3.2.2\n",
            "Cloning into 'cloneseer'...\n",
            "remote: Enumerating objects: 51, done.\u001b[K\n",
            "remote: Counting objects: 100% (51/51), done.\u001b[K\n",
            "remote: Compressing objects: 100% (51/51), done.\u001b[K\n",
            "remote: Total 51 (delta 21), reused 0 (delta 0), pack-reused 0 (from 0)\u001b[K\n",
            "Receiving objects: 100% (51/51), 5.45 MiB | 15.51 MiB/s, done.\n",
            "Resolving deltas: 100% (21/21), done.\n",
            "Collecting git+https://github.com/aerijman/igblast-parser.git\n",
            "  Cloning https://github.com/aerijman/igblast-parser.git to /tmp/pip-req-build-um9s8jf3\n",
            "  Running command git clone --filter=blob:none --quiet https://github.com/aerijman/igblast-parser.git /tmp/pip-req-build-um9s8jf3\n",
            "  Resolved https://github.com/aerijman/igblast-parser.git to commit 5ca74be20f3cf081cf0dd357a6eec57b04d3602e\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.12/dist-packages (from igblast_parser==0.0.4) (2.2.2)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.12/dist-packages (from igblast_parser==0.0.4) (2.0.2)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.12/dist-packages (from pandas->igblast_parser==0.0.4) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.12/dist-packages (from pandas->igblast_parser==0.0.4) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.12/dist-packages (from pandas->igblast_parser==0.0.4) (2025.2)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.12/dist-packages (from python-dateutil>=2.8.2->pandas->igblast_parser==0.0.4) (1.17.0)\n",
            "Building wheels for collected packages: igblast_parser\n",
            "  Building wheel for igblast_parser (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for igblast_parser: filename=igblast_parser-0.0.4-cp312-cp312-linux_x86_64.whl size=272248 sha256=165881318058591c4c0678e21e1941aac4d06466b20869f00a173c5ffa46b903\n",
            "  Stored in directory: /tmp/pip-ephem-wheel-cache-tq28kpg3/wheels/99/b7/d5/6b5f7bfc5f9d90dbff55a1853ad7ba194882890ed16d1d7824\n",
            "Successfully built igblast_parser\n",
            "Installing collected packages: igblast_parser\n",
            "Successfully installed igblast_parser-0.0.4\n",
            "Collecting fastcluster\n",
            "  Downloading fastcluster-1.3.0-cp312-cp312-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.7 kB)\n",
            "Requirement already satisfied: numpy>=2 in /usr/local/lib/python3.12/dist-packages (from fastcluster) (2.0.2)\n",
            "Downloading fastcluster-1.3.0-cp312-cp312-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (194 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m194.1/194.1 kB\u001b[0m \u001b[31m3.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: fastcluster\n",
            "Successfully installed fastcluster-1.3.0\n",
            "ncbi-igblast-1.22.0/\n",
            "ncbi-igblast-1.22.0/LICENSE\n",
            "ncbi-igblast-1.22.0/ncbi_package_info\n",
            "ncbi-igblast-1.22.0/ChangeLog\n",
            "ncbi-igblast-1.22.0/bin/\n",
            "ncbi-igblast-1.22.0/bin/makeogrannote.py\n",
            "ncbi-igblast-1.22.0/bin/igblastn\n",
            "ncbi-igblast-1.22.0/bin/makeblastdb\n",
            "ncbi-igblast-1.22.0/bin/blastdbcmd\n",
            "ncbi-igblast-1.22.0/bin/edit_imgt_file.pl\n",
            "ncbi-igblast-1.22.0/bin/igblastp\n",
            "ncbi-igblast-1.22.0/bin/makeogrdb.py\n",
            "ncbi-igblast-1.22.0/README\n",
            "ncbi-igblast-1.22.0/optional_file/\n",
            "ncbi-igblast-1.22.0/optional_file/mouse_D.frame\n",
            "ncbi-igblast-1.22.0/optional_file/mouse_gl.aux\n",
            "ncbi-igblast-1.22.0/optional_file/rabbit_gl.aux\n",
            "ncbi-igblast-1.22.0/optional_file/human_gl.aux.testonly\n",
            "ncbi-igblast-1.22.0/optional_file/rat_gl.aux\n",
            "ncbi-igblast-1.22.0/optional_file/human_gl.aux\n",
            "ncbi-igblast-1.22.0/optional_file/rhesus_monkey_gl.aux\n",
            "ncbi-igblast-1.22.0/internal_data/\n",
            "ncbi-igblast-1.22.0/internal_data/rat/\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.psi\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.pog\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat.pdm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat.ndm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.psd\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.nsi\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.nsd\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat.ndm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat.pdm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/human/\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.nos\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.ntf\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/human/human.pdm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.psd\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.nsi\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.psi\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.nto\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.psi\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.nsd\n",
            "ncbi-igblast-1.22.0/internal_data/human/human.ndm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.pog\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.pog\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/human/human.pdm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.not\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.psd\n",
            "ncbi-igblast-1.22.0/internal_data/human/human.ndm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.ndb\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pot\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.psi\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse.pdm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse.ndm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.psd\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.njs\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pdb\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.nsi\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.nsd\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse.ndm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pjs\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.ndb\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse.pdm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.ptf\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.pog\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pog\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.ntf\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pto\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.not\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.nos\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pos\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.nto\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.nsi\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.psd\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit.pdm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit.ndm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit.ndm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.nsd\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.psi\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.pog\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit.pdm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.ndb\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey.pdm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey.ndm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.not\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey.ndm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.pto\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey.pdm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.pdb\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.pot\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.nto\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.ntf\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.pos\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.nos\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.ptf\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.pog\n",
            "--2025-09-05 20:48:49--  https://www.imgt.org/download/V-QUEST/IMGT_V-QUEST_reference_directory/Homo_sapiens/IG/IGHV.fasta\n",
            "Resolving www.imgt.org (www.imgt.org)... 195.83.85.130\n",
            "Connecting to www.imgt.org (www.imgt.org)|195.83.85.130|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 190872 (186K) [text/plain]\n",
            "Saving to: ‘IGHV.fasta’\n",
            "\n",
            "IGHV.fasta          100%[===================>] 186.40K   409KB/s    in 0.5s    \n",
            "\n",
            "2025-09-05 20:48:51 (409 KB/s) - ‘IGHV.fasta’ saved [190872/190872]\n",
            "\n",
            "--2025-09-05 20:48:51--  https://www.imgt.org/download/V-QUEST/IMGT_V-QUEST_reference_directory/Homo_sapiens/IG/IGHD.fasta\n",
            "Resolving www.imgt.org (www.imgt.org)... 195.83.85.130\n",
            "Connecting to www.imgt.org (www.imgt.org)|195.83.85.130|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 5206 (5.1K) [text/plain]\n",
            "Saving to: ‘IGHD.fasta’\n",
            "\n",
            "IGHD.fasta          100%[===================>]   5.08K  --.-KB/s    in 0s      \n",
            "\n",
            "2025-09-05 20:48:51 (1.14 GB/s) - ‘IGHD.fasta’ saved [5206/5206]\n",
            "\n",
            "--2025-09-05 20:48:51--  https://www.imgt.org/download/V-QUEST/IMGT_V-QUEST_reference_directory/Homo_sapiens/IG/IGHJ.fasta\n",
            "Resolving www.imgt.org (www.imgt.org)... 195.83.85.130\n",
            "Connecting to www.imgt.org (www.imgt.org)|195.83.85.130|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 2043 (2.0K) [text/plain]\n",
            "Saving to: ‘IGHJ.fasta’\n",
            "\n",
            "IGHJ.fasta          100%[===================>]   2.00K  --.-KB/s    in 0s      \n",
            "\n",
            "2025-09-05 20:48:52 (722 MB/s) - ‘IGHJ.fasta’ saved [2043/2043]\n",
            "\n",
            "Archive:  cloneseer/Input_Folder_Rep1.zip\n",
            "   creating: Input_Folder_Rep1/\n",
            "  inflating: __MACOSX/._Input_Folder_Rep1  \n",
            "   creating: Input_Folder_Rep1/IGH_FR3_output/\n",
            "  inflating: __MACOSX/Input_Folder_Rep1/._IGH_FR3_output  \n",
            "   creating: Input_Folder_Rep1/IGH_FR1_output/\n",
            "  inflating: __MACOSX/Input_Folder_Rep1/._IGH_FR1_output  \n",
            "   creating: Input_Folder_Rep1/IGH_FR2_output/\n",
            "  inflating: __MACOSX/Input_Folder_Rep1/._IGH_FR2_output  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-080_S17_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-080_S17_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-081_S18_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-081_S18_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-080_S17_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-080_S17_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-081_S18_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-081_S18_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-080_S17_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-080_S17_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-081_S18_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-081_S18_L001_001_combined.fastq_unique_reads.tsv  \n",
            "Archive:  cloneseer/Input_Folder_Rep2.zip\n",
            "   creating: Input_Folder_Rep2/\n",
            "  inflating: __MACOSX/._Input_Folder_Rep2  \n",
            "   creating: Input_Folder_Rep2/IGH_FR3_output/\n",
            "  inflating: __MACOSX/Input_Folder_Rep2/._IGH_FR3_output  \n",
            "   creating: Input_Folder_Rep2/IGH_FR1_output/\n",
            "  inflating: __MACOSX/Input_Folder_Rep2/._IGH_FR1_output  \n",
            "   creating: Input_Folder_Rep2/IGH_FR2_output/\n",
            "  inflating: __MACOSX/Input_Folder_Rep2/._IGH_FR2_output  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-081_S12_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-081_S12_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-080_S13_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-080_S13_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-081_S12_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-081_S12_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-080_S13_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-080_S13_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-081_S12_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-081_S12_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-080_S13_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-080_S13_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "Collecting streamlit==1.23.0\n",
            "  Downloading streamlit-1.23.0-py2.py3-none-any.whl.metadata (7.4 kB)\n",
            "Requirement already satisfied: altair<6,>=4.0 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (5.5.0)\n",
            "Requirement already satisfied: blinker<2,>=1.0.0 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (1.9.0)\n",
            "Requirement already satisfied: cachetools<6,>=4.0 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (5.5.2)\n",
            "Requirement already satisfied: click<9,>=7.0 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (8.2.1)\n",
            "Collecting importlib-metadata<7,>=1.4 (from streamlit==1.23.0)\n",
            "  Downloading importlib_metadata-6.11.0-py3-none-any.whl.metadata (4.9 kB)\n",
            "Collecting numpy<2,>=1 (from streamlit==1.23.0)\n",
            "  Downloading numpy-1.26.4-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (61 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m61.0/61.0 kB\u001b[0m \u001b[31m3.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting packaging<24,>=14.1 (from streamlit==1.23.0)\n",
            "  Downloading packaging-23.2-py3-none-any.whl.metadata (3.2 kB)\n",
            "Requirement already satisfied: pandas<3,>=0.25 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (2.2.2)\n",
            "Collecting pillow<10,>=6.2.0 (from streamlit==1.23.0)\n",
            "  Downloading Pillow-9.5.0.tar.gz (50.5 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m50.5/50.5 MB\u001b[0m \u001b[31m12.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting protobuf<5,>=3.20 (from streamlit==1.23.0)\n",
            "  Downloading protobuf-4.25.8-cp37-abi3-manylinux2014_x86_64.whl.metadata (541 bytes)\n",
            "Requirement already satisfied: pyarrow>=4.0 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (18.1.0)\n",
            "Collecting pympler<2,>=0.9 (from streamlit==1.23.0)\n",
            "  Downloading Pympler-1.1-py3-none-any.whl.metadata (3.6 kB)\n",
            "Requirement already satisfied: python-dateutil<3,>=2 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (2.9.0.post0)\n",
            "Requirement already satisfied: requests<3,>=2.4 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (2.32.4)\n",
            "Requirement already satisfied: rich<14,>=10.11.0 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (13.9.4)\n",
            "Requirement already satisfied: tenacity<9,>=8.0.0 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (8.5.0)\n",
            "Requirement already satisfied: toml<2 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (0.10.2)\n",
            "Requirement already satisfied: typing-extensions>=3.10.0.0 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (4.15.0)\n",
            "Collecting tzlocal<5,>=1.1 (from streamlit==1.23.0)\n",
            "  Downloading tzlocal-4.3.1-py3-none-any.whl.metadata (14 kB)\n",
            "Collecting validators<1,>=0.2 (from streamlit==1.23.0)\n",
            "  Downloading validators-0.35.0-py3-none-any.whl.metadata (3.9 kB)\n",
            "Requirement already satisfied: gitpython!=3.1.19,<4,>=3 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (3.1.45)\n",
            "Collecting pydeck<1,>=0.1.dev5 (from streamlit==1.23.0)\n",
            "  Downloading pydeck-0.9.1-py2.py3-none-any.whl.metadata (4.1 kB)\n",
            "Requirement already satisfied: tornado<7,>=6.0.3 in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (6.4.2)\n",
            "Requirement already satisfied: watchdog in /usr/local/lib/python3.12/dist-packages (from streamlit==1.23.0) (6.0.0)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.12/dist-packages (from altair<6,>=4.0->streamlit==1.23.0) (3.1.6)\n",
            "Requirement already satisfied: jsonschema>=3.0 in /usr/local/lib/python3.12/dist-packages (from altair<6,>=4.0->streamlit==1.23.0) (4.25.1)\n",
            "Requirement already satisfied: narwhals>=1.14.2 in /usr/local/lib/python3.12/dist-packages (from altair<6,>=4.0->streamlit==1.23.0) (2.3.0)\n",
            "Requirement already satisfied: gitdb<5,>=4.0.1 in /usr/local/lib/python3.12/dist-packages (from gitpython!=3.1.19,<4,>=3->streamlit==1.23.0) (4.0.12)\n",
            "Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.12/dist-packages (from importlib-metadata<7,>=1.4->streamlit==1.23.0) (3.23.0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.12/dist-packages (from pandas<3,>=0.25->streamlit==1.23.0) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.12/dist-packages (from pandas<3,>=0.25->streamlit==1.23.0) (2025.2)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.12/dist-packages (from python-dateutil<3,>=2->streamlit==1.23.0) (1.17.0)\n",
            "Requirement already satisfied: charset_normalizer<4,>=2 in /usr/local/lib/python3.12/dist-packages (from requests<3,>=2.4->streamlit==1.23.0) (3.4.3)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.12/dist-packages (from requests<3,>=2.4->streamlit==1.23.0) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.12/dist-packages (from requests<3,>=2.4->streamlit==1.23.0) (2.5.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.12/dist-packages (from requests<3,>=2.4->streamlit==1.23.0) (2025.8.3)\n",
            "Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.12/dist-packages (from rich<14,>=10.11.0->streamlit==1.23.0) (4.0.0)\n",
            "Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.12/dist-packages (from rich<14,>=10.11.0->streamlit==1.23.0) (2.19.2)\n",
            "Collecting pytz-deprecation-shim (from tzlocal<5,>=1.1->streamlit==1.23.0)\n",
            "  Downloading pytz_deprecation_shim-0.1.0.post0-py2.py3-none-any.whl.metadata (5.2 kB)\n",
            "Requirement already satisfied: smmap<6,>=3.0.1 in /usr/local/lib/python3.12/dist-packages (from gitdb<5,>=4.0.1->gitpython!=3.1.19,<4,>=3->streamlit==1.23.0) (5.0.2)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.12/dist-packages (from jinja2->altair<6,>=4.0->streamlit==1.23.0) (3.0.2)\n",
            "Requirement already satisfied: attrs>=22.2.0 in /usr/local/lib/python3.12/dist-packages (from jsonschema>=3.0->altair<6,>=4.0->streamlit==1.23.0) (25.3.0)\n",
            "Requirement already satisfied: jsonschema-specifications>=2023.03.6 in /usr/local/lib/python3.12/dist-packages (from jsonschema>=3.0->altair<6,>=4.0->streamlit==1.23.0) (2025.4.1)\n",
            "Requirement already satisfied: referencing>=0.28.4 in /usr/local/lib/python3.12/dist-packages (from jsonschema>=3.0->altair<6,>=4.0->streamlit==1.23.0) (0.36.2)\n",
            "Requirement already satisfied: rpds-py>=0.7.1 in /usr/local/lib/python3.12/dist-packages (from jsonschema>=3.0->altair<6,>=4.0->streamlit==1.23.0) (0.27.1)\n",
            "Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.12/dist-packages (from markdown-it-py>=2.2.0->rich<14,>=10.11.0->streamlit==1.23.0) (0.1.2)\n",
            "Downloading streamlit-1.23.0-py2.py3-none-any.whl (8.9 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m8.9/8.9 MB\u001b[0m \u001b[31m82.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading importlib_metadata-6.11.0-py3-none-any.whl (23 kB)\n",
            "Downloading numpy-1.26.4-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.0 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m18.0/18.0 MB\u001b[0m \u001b[31m27.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading packaging-23.2-py3-none-any.whl (53 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m53.0/53.0 kB\u001b[0m \u001b[31m4.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading protobuf-4.25.8-cp37-abi3-manylinux2014_x86_64.whl (294 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m294.9/294.9 kB\u001b[0m \u001b[31m23.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pydeck-0.9.1-py2.py3-none-any.whl (6.9 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m6.9/6.9 MB\u001b[0m \u001b[31m109.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading Pympler-1.1-py3-none-any.whl (165 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m165.8/165.8 kB\u001b[0m \u001b[31m11.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading tzlocal-4.3.1-py3-none-any.whl (20 kB)\n",
            "Downloading validators-0.35.0-py3-none-any.whl (44 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m44.7/44.7 kB\u001b[0m \u001b[31m2.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pytz_deprecation_shim-0.1.0.post0-py2.py3-none-any.whl (15 kB)\n",
            "Building wheels for collected packages: pillow\n",
            "  Building wheel for pillow (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pillow: filename=Pillow-9.5.0-cp312-cp312-linux_x86_64.whl size=1210267 sha256=fc9da0127ae229c7d92a7e2914f1e91ce9118fd36a6a0e2c50e5f4f5e0842f2c\n",
            "  Stored in directory: /root/.cache/pip/wheels/ea/de/2e/75a6399e5d8cd3a55c13c8f0658d996d4ce4cff37389de044c\n",
            "Successfully built pillow\n",
            "Installing collected packages: validators, pytz-deprecation-shim, pympler, protobuf, pillow, packaging, numpy, importlib-metadata, tzlocal, pydeck, streamlit\n",
            "  Attempting uninstall: protobuf\n",
            "    Found existing installation: protobuf 5.29.5\n",
            "    Uninstalling protobuf-5.29.5:\n",
            "      Successfully uninstalled protobuf-5.29.5\n",
            "  Attempting uninstall: pillow\n",
            "    Found existing installation: pillow 11.3.0\n",
            "    Uninstalling pillow-11.3.0:\n",
            "      Successfully uninstalled pillow-11.3.0\n",
            "  Attempting uninstall: packaging\n",
            "    Found existing installation: packaging 25.0\n",
            "    Uninstalling packaging-25.0:\n",
            "      Successfully uninstalled packaging-25.0\n",
            "  Attempting uninstall: numpy\n",
            "    Found existing installation: numpy 2.0.2\n",
            "    Uninstalling numpy-2.0.2:\n",
            "      Successfully uninstalled numpy-2.0.2\n",
            "  Attempting uninstall: importlib-metadata\n",
            "    Found existing installation: importlib_metadata 8.7.0\n",
            "    Uninstalling importlib_metadata-8.7.0:\n",
            "      Successfully uninstalled importlib_metadata-8.7.0\n",
            "  Attempting uninstall: tzlocal\n",
            "    Found existing installation: tzlocal 5.3.1\n",
            "    Uninstalling tzlocal-5.3.1:\n",
            "      Successfully uninstalled tzlocal-5.3.1\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "fastcluster 1.3.0 requires numpy>=2, but you have numpy 1.26.4 which is incompatible.\n",
            "opencv-contrib-python 4.12.0.88 requires numpy<2.3.0,>=2; python_version >= \"3.9\", but you have numpy 1.26.4 which is incompatible.\n",
            "opencv-python-headless 4.12.0.88 requires numpy<2.3.0,>=2; python_version >= \"3.9\", but you have numpy 1.26.4 which is incompatible.\n",
            "grpcio-status 1.71.2 requires protobuf<6.0dev,>=5.26.1, but you have protobuf 4.25.8 which is incompatible.\n",
            "opencv-python 4.12.0.88 requires numpy<2.3.0,>=2; python_version >= \"3.9\", but you have numpy 1.26.4 which is incompatible.\n",
            "ydf 0.13.0 requires protobuf<7.0.0,>=5.29.1, but you have protobuf 4.25.8 which is incompatible.\n",
            "google-cloud-bigquery 3.36.0 requires packaging>=24.2.0, but you have packaging 23.2 which is incompatible.\n",
            "scikit-image 0.25.2 requires pillow>=10.1, but you have pillow 9.5.0 which is incompatible.\n",
            "xarray 2025.8.0 requires packaging>=24.1, but you have packaging 23.2 which is incompatible.\n",
            "google-adk 1.13.0 requires tzlocal<6.0,>=5.3, but you have tzlocal 4.3.1 which is incompatible.\n",
            "thinc 8.3.6 requires numpy<3.0.0,>=2.0.0, but you have numpy 1.26.4 which is incompatible.\n",
            "db-dtypes 1.4.3 requires packaging>=24.2.0, but you have packaging 23.2 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSuccessfully installed importlib-metadata-6.11.0 numpy-1.26.4 packaging-23.2 pillow-9.5.0 protobuf-4.25.8 pydeck-0.9.1 pympler-1.1 pytz-deprecation-shim-0.1.0.post0 streamlit-1.23.0 tzlocal-4.3.1 validators-0.35.0\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "google",
                  "numpy",
                  "packaging"
                ]
              },
              "id": "93ece8b042e047e29bf2009ddf584eda"
            }
          },
          "metadata": {}
        }
      ],
      "source": [
        "!wget https://ftp.ncbi.nih.gov/blast/executables/igblast/release/LATEST/ncbi-igblast-1.22.0-x64-linux.tar.gz\n",
        "!pip install Bio\n",
        "!git clone https://github.com/Eitan177/cloneseer.git\n",
        "!pip install git+https://github.com/aerijman/igblast-parser.git\n",
        "!pip install fastcluster\n",
        "!export IGDATA=\"/content/ncbi-igblast-1.22.0/bin\"\n",
        "!tar -xzvf ncbi-igblast-1.22.0-x64-linux.tar.gz\n",
        "import sys\n",
        "import subprocess\n",
        "import pandas as pd\n",
        "from io import StringIO\n",
        "import warnings\n",
        "import numpy as np\n",
        "from Bio import SeqIO\n",
        "import os\n",
        "os.chdir('/content/ncbi-igblast-1.22.0')\n",
        "os.mkdir('database')\n",
        "os.mkdir('database_clean')\n",
        "os.chdir('/content/ncbi-igblast-1.22.0/database')\n",
        "!wget https://www.imgt.org/download/V-QUEST/IMGT_V-QUEST_reference_directory/Homo_sapiens/IG/IGHV.fasta\n",
        "!wget https://www.imgt.org/download/V-QUEST/IMGT_V-QUEST_reference_directory/Homo_sapiens/IG/IGHD.fasta\n",
        "!wget https://www.imgt.org/download/V-QUEST/IMGT_V-QUEST_reference_directory/Homo_sapiens/IG/IGHJ.fasta\n",
        "os.chdir('/content/ncbi-igblast-1.22.0')\n",
        "# cleaning up IMGT germline sequence files\n",
        "input_imgt_ref = 'database/IGHV.fasta'\n",
        "#input_imgt_ref =\"/content/ighv_functional.fasta\"\n",
        "output_imgt_ref = 'database_clean/IGHV_clean.fasta'\n",
        "\n",
        "cmd = ['bin/edit_imgt_file.pl', input_imgt_ref , '>', output_imgt_ref]\n",
        "\n",
        "\n",
        "fh = open(output_imgt_ref, 'w')\n",
        "a = subprocess.Popen(cmd, stdout=fh)\n",
        "a.wait()\n",
        "\n",
        "input_imgt_ref = 'database/IGHD.fasta'\n",
        "output_imgt_ref = 'database_clean/IGHD_clean.fasta'\n",
        "\n",
        "cmd = ['bin/edit_imgt_file.pl', input_imgt_ref , '>', output_imgt_ref]\n",
        "\n",
        "fh = open(output_imgt_ref, 'w')\n",
        "a = subprocess.Popen(cmd, stdout=fh)\n",
        "a.wait()\n",
        "input_imgt_ref = 'database/IGHJ.fasta'\n",
        "output_imgt_ref = 'database_clean/IGHJ_clean.fasta'\n",
        "\n",
        "cmd = ['bin/edit_imgt_file.pl', input_imgt_ref , '>', output_imgt_ref]\n",
        "\n",
        "# display result (and any error) in notebook. No file will be saved\n",
        "# subprocess.run(cmd.split(), capture_output=True)\n",
        "\n",
        "# save cmd output to file\n",
        "fh = open(output_imgt_ref, 'w')\n",
        "a = subprocess.Popen(cmd, stdout=fh)\n",
        "a.wait()\n",
        "cmd = ['bin/makeblastdb', '-parse_seqids', '-dbtype', 'nucl', '-in', '/content/ncbi-igblast-1.22.0/database_clean/IGHD_clean.fasta']\n",
        "\n",
        "a = subprocess.Popen(cmd, stdout=subprocess.PIPE, cwd='/content/ncbi-igblast-1.22.0')\n",
        "a.wait()\n",
        "cmd = ['bin/makeblastdb', '-parse_seqids', '-dbtype', 'nucl', '-in', '/content/ncbi-igblast-1.22.0/database_clean/IGHV_clean.fasta']\n",
        "\n",
        "a = subprocess.Popen(cmd, stdout=subprocess.PIPE, cwd='/content/ncbi-igblast-1.22.0')\n",
        "a.wait()\n",
        "cmd = ['bin/makeblastdb', '-parse_seqids', '-dbtype', 'nucl', '-in', '/content/ncbi-igblast-1.22.0/database_clean/IGHJ_clean.fasta']\n",
        "\n",
        "a = subprocess.Popen(cmd, stdout=subprocess.PIPE, cwd='/content/ncbi-igblast-1.22.0')\n",
        "a.wait()\n",
        "\n",
        "os.chdir('/content/')\n",
        "!mv /content/cloneseer/R14-078.xlsx /content/R14-078.xlsx\n",
        "!mv /content/cloneseer/R14-079.xlsx /content/R14-079.xlsx\n",
        "!mkdir /content/cloneseer_out/\n",
        "!unzip cloneseer/Input_Folder_Rep1.zip\n",
        "!unzip cloneseer/Input_Folder_Rep2.zip\n",
        "!pip install streamlit==1.23.0\n",
        "os.chdir('/content/ncbi-igblast-1.22.0')\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!export IGDATA=\"/content/ncbi-igblast-1.22.0/bin\"\n",
        "import sys\n",
        "import subprocess\n",
        "import pandas as pd\n",
        "from io import StringIO\n",
        "import warnings\n",
        "import numpy as np\n",
        "from Bio import SeqIO\n",
        "import os\n",
        "os.chdir('/content/ncbi-igblast-1.22.0')\n",
        "!python /content/cloneseer/lymphotrack_main_local.py --sample_sheet /content/cloneseer/R14-078_R14-079_samplesheet.xlsx --clone_length_fr1 40 --clone_length_fr2 30 --clone_length_fr3 20 --independent_kmer 1 --must_be_identical 0.95 --save_path /content/cloneseer_out --skiprows 0 --AnalyzePositiveControl 0 --Match_Clone_with_FR 0 --readrows 50 --Input_Nanograms 50 --aggregate 0"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zIgVl6ulhd1y",
        "outputId": "5a07bea1-5013-4327-a645-d715b9e6d34d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/usr/local/lib/python3.12/dist-packages/Bio/pairwise2.py:278: BiopythonDeprecationWarning: Bio.pairwise2 has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.PairwiseAligner as a replacement, and contact the Biopython developers if you still need the Bio.pairwise2 module.\n",
            "  warnings.warn(\n",
            "2025-09-05 20:51:16.385 \n",
            "  \u001b[33m\u001b[1mWarning:\u001b[0m to view this Streamlit app on a browser, run it with the following\n",
            "  command:\n",
            "\n",
            "    streamlit run /content/cloneseer/lymphotrack_main_local.py [ARGUMENTS]\n",
            "2025-09-05 20:51:16.386 `st.experimental_memo` is deprecated. Please use the new command `st.cache_data` instead, which has the same behavior. More information [in our docs](https://docs.streamlit.io/library/advanced-features/caching).\n",
            "2025-09-05 20:51:16.386 No runtime found, using MemoryCacheStorageManager\n",
            "/content/cloneseer/multiple_sequence_alignment.py:105: SyntaxWarning: invalid escape sequence '\\='\n",
            "  if(re.findall('^Query\\=',line) != []):\n",
            "/content/cloneseer/multiple_sequence_alignment.py:115: SyntaxWarning: invalid escape sequence '\\/'\n",
            "  percentline.append(['','',re.sub('^.+\\/','',re.sub('\\)','',line.split(' ')[3])),re.sub('\\/.+','',re.sub('\\(','',line.split(' ')[3])), '','',  re.sub('%','',line.split(' ')[2])])\n",
            "/content/cloneseer/multiple_sequence_alignment.py:115: SyntaxWarning: invalid escape sequence '\\)'\n",
            "  percentline.append(['','',re.sub('^.+\\/','',re.sub('\\)','',line.split(' ')[3])),re.sub('\\/.+','',re.sub('\\(','',line.split(' ')[3])), '','',  re.sub('%','',line.split(' ')[2])])\n",
            "/content/cloneseer/multiple_sequence_alignment.py:115: SyntaxWarning: invalid escape sequence '\\/'\n",
            "  percentline.append(['','',re.sub('^.+\\/','',re.sub('\\)','',line.split(' ')[3])),re.sub('\\/.+','',re.sub('\\(','',line.split(' ')[3])), '','',  re.sub('%','',line.split(' ')[2])])\n",
            "/content/cloneseer/multiple_sequence_alignment.py:115: SyntaxWarning: invalid escape sequence '\\('\n",
            "  percentline.append(['','',re.sub('^.+\\/','',re.sub('\\)','',line.split(' ')[3])),re.sub('\\/.+','',re.sub('\\(','',line.split(' ')[3])), '','',  re.sub('%','',line.split(' ')[2])])\n",
            "/content/cloneseer/multiple_sequence_alignment.py:117: SyntaxWarning: invalid escape sequence '\\='\n",
            "  querysequencename=[re.sub('^.+_','',re.sub('^Query\\= ','',x)) for x in querysequencename]\n",
            "2025-09-05 20:51:17.012 `st.experimental_memo` is deprecated. Please use the new command `st.cache_data` instead, which has the same behavior. More information [in our docs](https://docs.streamlit.io/library/advanced-features/caching).\n",
            "2025-09-05 20:51:17.013 No runtime found, using MemoryCacheStorageManager\n",
            "argument parsing\n",
            "0\n",
            "Sample ID Path      /content/Input_Folder_Rep2\n",
            "Sample ID                              R14-078\n",
            "Replicate Number                             2\n",
            "Clone File Path                      /content/\n",
            "Clone File                        R14-078.xlsx\n",
            "Descriptors                                NaN\n",
            "Name: 0, dtype: object\n",
            "about to process sample sheet\n",
            "made dictionary\n",
            "R14-078_FR1_Clone1\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n",
            "Framework 1\n",
            "Warning: Auxilary data file could not be found\n",
            "2025-09-05 20:51:22.139 No runtime found, using MemoryCacheStorageManager\n",
            "17\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1073', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1067', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1155', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1161', ...)}\n",
            "Warning: Auxilary data file could not be found\n",
            "N/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCACCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTTTGACCGCTGCGTACACTGCCGTGTATTACTTTTCTAGAGACTACAAGAACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGATCTCCCTGAACCTGAGCTCTGTGACCGCTGCGTACACTGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTAACCACTGCTTACACGTCCGTTTATTACTTTTCGAGAGAGTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGAGGACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGAAACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTGCGAGACACTACAATTACGACGGGATGGACGTCTGGGGCCAAN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTGCTAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTTCGAGAGACTACAACTACTACGGTATGGTCGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTTCTAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTTTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTTTTCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCTTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGTACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCTGACACGGCCGTGTATTACTGTTCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGCCTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGAGAACACGGCAGAGTATTACTATGTGAGAGACTACACCTCCTACGGTATGGACGTCTGGGGCCAAGGGACCACERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text_font='monospace' [no close matches] {renderer: GlyphRenderer(id='p1241', ...)}\n",
            "/content/cloneseer/view_alignments_0401.py:287: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
            "  table_w_sequence_count_sorted_w_filtered_vdj = table_w_sequence_count_sorted_w_filtered_vdj[table_w_sequence_count_sorted_w_filtered_vdj[['mother_clone_distance','_merge']].apply(lambda x: x[0]<3 or x[1]=='both',axis=1)].drop(columns=[\"_merge\"])\n",
            "/content/cloneseer/view_alignments_0401.py:293: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
            "  table_w_sequence_count_sorted_w_filtered_vdj= table_w_sequence_count_sorted_w_filtered_vdj[table_w_sequence_count_sorted_w_filtered_vdj[['top_v_gene_match_x','top_v_gene_match_y','v_end_x','v_end_y','j_start_x','j_start_y','insert_trinucleotide','delete_trinucleotide','mother_clone_distance']].apply(lambda x: (re.sub(';.+','',str(x[0]))==re.sub(';.+','',str(x[1])) and str(x[2])==str(x[3]) and str(x[4])==str(x[5])) or (str(x[0])==str(x[1]) and ((x[6] % 3 == 0 and x[6]>2) or (x[7] % 3 == 0 and x[7]>2)) or (x[8]<3)),axis=1)]\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1326', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1320', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1408', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1414', ...)}\n",
            "R14-078_FR1_Clone1\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n"
          ]
        }
      ]
    }
  ]
}