{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyOKmeknass9vmaSWjIPrN3w",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Eitan177/cloneseer/blob/main/0904example_usage.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "cgRU4h5-gxmy",
        "outputId": "b9689923-23b7-4e93-d2ca-f89972c7b8cb"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-09-05 20:18:00--  https://ftp.ncbi.nih.gov/blast/executables/igblast/release/LATEST/ncbi-igblast-1.22.0-x64-linux.tar.gz\n",
            "Resolving ftp.ncbi.nih.gov (ftp.ncbi.nih.gov)... 130.14.250.13, 130.14.250.7, 130.14.250.10, ...\n",
            "Connecting to ftp.ncbi.nih.gov (ftp.ncbi.nih.gov)|130.14.250.13|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 46009645 (44M) [application/x-gzip]\n",
            "Saving to: ‘ncbi-igblast-1.22.0-x64-linux.tar.gz’\n",
            "\n",
            "ncbi-igblast-1.22.0 100%[===================>]  43.88M  26.3MB/s    in 1.7s    \n",
            "\n",
            "2025-09-05 20:18:02 (26.3 MB/s) - ‘ncbi-igblast-1.22.0-x64-linux.tar.gz’ saved [46009645/46009645]\n",
            "\n",
            "Collecting Bio\n",
            "  Downloading bio-1.8.0-py3-none-any.whl.metadata (5.7 kB)\n",
            "Collecting biopython>=1.80 (from Bio)\n",
            "  Downloading biopython-1.85-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (13 kB)\n",
            "Collecting gprofiler-official (from Bio)\n",
            "  Downloading gprofiler_official-1.0.0-py3-none-any.whl.metadata (11 kB)\n",
            "Collecting mygene (from Bio)\n",
            "  Downloading mygene-3.2.2-py2.py3-none-any.whl.metadata (10 kB)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.12/dist-packages (from Bio) (2.2.2)\n",
            "Requirement already satisfied: pooch in /usr/local/lib/python3.12/dist-packages (from Bio) (1.8.2)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.12/dist-packages (from Bio) (2.32.4)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.12/dist-packages (from Bio) (4.67.1)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.12/dist-packages (from biopython>=1.80->Bio) (2.0.2)\n",
            "Collecting biothings-client>=0.2.6 (from mygene->Bio)\n",
            "  Downloading biothings_client-0.4.1-py3-none-any.whl.metadata (10 kB)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.12/dist-packages (from pandas->Bio) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.12/dist-packages (from pandas->Bio) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.12/dist-packages (from pandas->Bio) (2025.2)\n",
            "Requirement already satisfied: platformdirs>=2.5.0 in /usr/local/lib/python3.12/dist-packages (from pooch->Bio) (4.4.0)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.12/dist-packages (from pooch->Bio) (25.0)\n",
            "Requirement already satisfied: charset_normalizer<4,>=2 in /usr/local/lib/python3.12/dist-packages (from requests->Bio) (3.4.3)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.12/dist-packages (from requests->Bio) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.12/dist-packages (from requests->Bio) (2.5.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.12/dist-packages (from requests->Bio) (2025.8.3)\n",
            "Requirement already satisfied: httpx>=0.22.0 in /usr/local/lib/python3.12/dist-packages (from biothings-client>=0.2.6->mygene->Bio) (0.28.1)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.12/dist-packages (from python-dateutil>=2.8.2->pandas->Bio) (1.17.0)\n",
            "Requirement already satisfied: anyio in /usr/local/lib/python3.12/dist-packages (from httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (4.10.0)\n",
            "Requirement already satisfied: httpcore==1.* in /usr/local/lib/python3.12/dist-packages (from httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (1.0.9)\n",
            "Requirement already satisfied: h11>=0.16 in /usr/local/lib/python3.12/dist-packages (from httpcore==1.*->httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (0.16.0)\n",
            "Requirement already satisfied: sniffio>=1.1 in /usr/local/lib/python3.12/dist-packages (from anyio->httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (1.3.1)\n",
            "Requirement already satisfied: typing_extensions>=4.5 in /usr/local/lib/python3.12/dist-packages (from anyio->httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (4.15.0)\n",
            "Downloading bio-1.8.0-py3-none-any.whl (321 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m321.1/321.1 kB\u001b[0m \u001b[31m12.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading biopython-1.85-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.3/3.3 MB\u001b[0m \u001b[31m72.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading gprofiler_official-1.0.0-py3-none-any.whl (9.3 kB)\n",
            "Downloading mygene-3.2.2-py2.py3-none-any.whl (5.4 kB)\n",
            "Downloading biothings_client-0.4.1-py3-none-any.whl (46 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m46.7/46.7 kB\u001b[0m \u001b[31m3.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: biopython, gprofiler-official, biothings-client, mygene, Bio\n",
            "Successfully installed Bio-1.8.0 biopython-1.85 biothings-client-0.4.1 gprofiler-official-1.0.0 mygene-3.2.2\n",
            "Cloning into 'cloneseer'...\n",
            "remote: Enumerating objects: 40, done.\u001b[K\n",
            "remote: Counting objects: 100% (40/40), done.\u001b[K\n",
            "remote: Compressing objects: 100% (40/40), done.\u001b[K\n",
            "remote: Total 40 (delta 13), reused 0 (delta 0), pack-reused 0 (from 0)\u001b[K\n",
            "Receiving objects: 100% (40/40), 5.43 MiB | 20.23 MiB/s, done.\n",
            "Resolving deltas: 100% (13/13), done.\n",
            "Collecting git+https://github.com/aerijman/igblast-parser.git\n",
            "  Cloning https://github.com/aerijman/igblast-parser.git to /tmp/pip-req-build-hfkfey3m\n",
            "  Running command git clone --filter=blob:none --quiet https://github.com/aerijman/igblast-parser.git /tmp/pip-req-build-hfkfey3m\n",
            "  Resolved https://github.com/aerijman/igblast-parser.git to commit 5ca74be20f3cf081cf0dd357a6eec57b04d3602e\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.12/dist-packages (from igblast_parser==0.0.4) (2.2.2)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.12/dist-packages (from igblast_parser==0.0.4) (2.0.2)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.12/dist-packages (from pandas->igblast_parser==0.0.4) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.12/dist-packages (from pandas->igblast_parser==0.0.4) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.12/dist-packages (from pandas->igblast_parser==0.0.4) (2025.2)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.12/dist-packages (from python-dateutil>=2.8.2->pandas->igblast_parser==0.0.4) (1.17.0)\n",
            "Building wheels for collected packages: igblast_parser\n",
            "  Building wheel for igblast_parser (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for igblast_parser: filename=igblast_parser-0.0.4-cp312-cp312-linux_x86_64.whl size=272250 sha256=63304178c5aeafa1f1558c0d9bdd2633336569560dd375cbf8329a9209c56671\n",
            "  Stored in directory: /tmp/pip-ephem-wheel-cache-opeijpn2/wheels/99/b7/d5/6b5f7bfc5f9d90dbff55a1853ad7ba194882890ed16d1d7824\n",
            "Successfully built igblast_parser\n",
            "Installing collected packages: igblast_parser\n",
            "Successfully installed igblast_parser-0.0.4\n",
            "Collecting fastcluster\n",
            "  Downloading fastcluster-1.3.0-cp312-cp312-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.7 kB)\n",
            "Requirement already satisfied: numpy>=2 in /usr/local/lib/python3.12/dist-packages (from fastcluster) (2.0.2)\n",
            "Downloading fastcluster-1.3.0-cp312-cp312-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (194 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m194.1/194.1 kB\u001b[0m \u001b[31m6.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: fastcluster\n",
            "Successfully installed fastcluster-1.3.0\n",
            "ncbi-igblast-1.22.0/\n",
            "ncbi-igblast-1.22.0/LICENSE\n",
            "ncbi-igblast-1.22.0/ncbi_package_info\n",
            "ncbi-igblast-1.22.0/ChangeLog\n",
            "ncbi-igblast-1.22.0/bin/\n",
            "ncbi-igblast-1.22.0/bin/makeogrannote.py\n",
            "ncbi-igblast-1.22.0/bin/igblastn\n",
            "ncbi-igblast-1.22.0/bin/makeblastdb\n",
            "ncbi-igblast-1.22.0/bin/blastdbcmd\n",
            "ncbi-igblast-1.22.0/bin/edit_imgt_file.pl\n",
            "ncbi-igblast-1.22.0/bin/igblastp\n",
            "ncbi-igblast-1.22.0/bin/makeogrdb.py\n",
            "ncbi-igblast-1.22.0/README\n",
            "ncbi-igblast-1.22.0/optional_file/\n",
            "ncbi-igblast-1.22.0/optional_file/mouse_D.frame\n",
            "ncbi-igblast-1.22.0/optional_file/mouse_gl.aux\n",
            "ncbi-igblast-1.22.0/optional_file/rabbit_gl.aux\n",
            "ncbi-igblast-1.22.0/optional_file/human_gl.aux.testonly\n",
            "ncbi-igblast-1.22.0/optional_file/rat_gl.aux\n",
            "ncbi-igblast-1.22.0/optional_file/human_gl.aux\n",
            "ncbi-igblast-1.22.0/optional_file/rhesus_monkey_gl.aux\n",
            "ncbi-igblast-1.22.0/internal_data/\n",
            "ncbi-igblast-1.22.0/internal_data/rat/\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.psi\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.pog\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat.pdm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat.ndm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.psd\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.nsi\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat_V.nsd\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat.ndm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/rat/rat.pdm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/human/\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.nos\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.ntf\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/human/human.pdm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.psd\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.nsi\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.psi\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.nto\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.psi\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.nsd\n",
            "ncbi-igblast-1.22.0/internal_data/human/human.ndm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.pog\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.pog\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/human/human.pdm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.not\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.psd\n",
            "ncbi-igblast-1.22.0/internal_data/human/human.ndm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_V.ndb\n",
            "ncbi-igblast-1.22.0/internal_data/human/human_TR_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pot\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.psi\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse.pdm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse.ndm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.psd\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.njs\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pdb\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.nsi\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.nsd\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse.ndm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pjs\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.ndb\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse.pdm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.ptf\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.pog\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pog\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.ntf\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pto\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_TR_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.not\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.nos\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.pos\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/mouse/mouse_V.nto\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.nsi\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.psd\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit.pdm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit.ndm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit.ndm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.nsd\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.psi\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.pog\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit.pdm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/rabbit/rabbit_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.ndb\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey.pdm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey.ndm.kabat\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.not\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey.ndm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.psq\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.pto\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.nsq\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.nhr\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey.pdm.imgt\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.pdb\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.pot\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.nto\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.nin\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.ntf\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.pin\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.pos\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.nog\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.nos\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.ptf\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.phr\n",
            "ncbi-igblast-1.22.0/internal_data/rhesus_monkey/rhesus_monkey_V.pog\n",
            "--2025-09-05 20:18:42--  https://www.imgt.org/download/V-QUEST/IMGT_V-QUEST_reference_directory/Homo_sapiens/IG/IGHV.fasta\n",
            "Resolving www.imgt.org (www.imgt.org)... 195.83.85.130\n",
            "Connecting to www.imgt.org (www.imgt.org)|195.83.85.130|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 190872 (186K) [text/plain]\n",
            "Saving to: ‘IGHV.fasta’\n",
            "\n",
            "IGHV.fasta          100%[===================>] 186.40K  --.-KB/s    in 0.1s    \n",
            "\n",
            "2025-09-05 20:18:42 (1.81 MB/s) - ‘IGHV.fasta’ saved [190872/190872]\n",
            "\n",
            "--2025-09-05 20:18:42--  https://www.imgt.org/download/V-QUEST/IMGT_V-QUEST_reference_directory/Homo_sapiens/IG/IGHD.fasta\n",
            "Resolving www.imgt.org (www.imgt.org)... 195.83.85.130\n",
            "Connecting to www.imgt.org (www.imgt.org)|195.83.85.130|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 5206 (5.1K) [text/plain]\n",
            "Saving to: ‘IGHD.fasta’\n",
            "\n",
            "IGHD.fasta          100%[===================>]   5.08K  --.-KB/s    in 0s      \n",
            "\n",
            "2025-09-05 20:18:42 (1.51 GB/s) - ‘IGHD.fasta’ saved [5206/5206]\n",
            "\n",
            "--2025-09-05 20:18:42--  https://www.imgt.org/download/V-QUEST/IMGT_V-QUEST_reference_directory/Homo_sapiens/IG/IGHJ.fasta\n",
            "Resolving www.imgt.org (www.imgt.org)... 195.83.85.130\n",
            "Connecting to www.imgt.org (www.imgt.org)|195.83.85.130|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 2043 (2.0K) [text/plain]\n",
            "Saving to: ‘IGHJ.fasta’\n",
            "\n",
            "IGHJ.fasta          100%[===================>]   2.00K  --.-KB/s    in 0s      \n",
            "\n",
            "2025-09-05 20:18:43 (824 MB/s) - ‘IGHJ.fasta’ saved [2043/2043]\n",
            "\n",
            "Archive:  cloneseer/Input_Folder_Rep1.zip\n",
            "   creating: Input_Folder_Rep1/\n",
            "  inflating: __MACOSX/._Input_Folder_Rep1  \n",
            "   creating: Input_Folder_Rep1/IGH_FR3_output/\n",
            "  inflating: __MACOSX/Input_Folder_Rep1/._IGH_FR3_output  \n",
            "   creating: Input_Folder_Rep1/IGH_FR1_output/\n",
            "  inflating: __MACOSX/Input_Folder_Rep1/._IGH_FR1_output  \n",
            "   creating: Input_Folder_Rep1/IGH_FR2_output/\n",
            "  inflating: __MACOSX/Input_Folder_Rep1/._IGH_FR2_output  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-080_S17_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-080_S17_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR3_output/R14-081_S18_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR3_output/._R14-081_S18_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-080_S17_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-080_S17_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR1_output/R14-081_S18_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR1_output/._R14-081_S18_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-080_S17_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-080_S17_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep1/IGH_FR2_output/R14-081_S18_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep1/IGH_FR2_output/._R14-081_S18_L001_001_combined.fastq_unique_reads.tsv  \n",
            "Archive:  cloneseer/Input_Folder_Rep2.zip\n",
            "   creating: Input_Folder_Rep2/\n",
            "  inflating: __MACOSX/._Input_Folder_Rep2  \n",
            "   creating: Input_Folder_Rep2/IGH_FR3_output/\n",
            "  inflating: __MACOSX/Input_Folder_Rep2/._IGH_FR3_output  \n",
            "   creating: Input_Folder_Rep2/IGH_FR1_output/\n",
            "  inflating: __MACOSX/Input_Folder_Rep2/._IGH_FR1_output  \n",
            "   creating: Input_Folder_Rep2/IGH_FR2_output/\n",
            "  inflating: __MACOSX/Input_Folder_Rep2/._IGH_FR2_output  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-081_S12_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-081_S12_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-080_S13_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-080_S13_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR3_output/R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR3_output/._R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-081_S12_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-081_S12_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-080_S13_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-080_S13_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR1_output/R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR1_output/._R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-117_S21_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-081_S12_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-081_S12_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-118_S22_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-079_S15_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-119_S23_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-080_S13_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-080_S13_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-078_S14_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: Input_Folder_Rep2/IGH_FR2_output/R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n",
            "  inflating: __MACOSX/Input_Folder_Rep2/IGH_FR2_output/._R14-120_S24_L001_001_combined.fastq_unique_reads.tsv  \n"
          ]
        }
      ],
      "source": [
        "!wget https://ftp.ncbi.nih.gov/blast/executables/igblast/release/LATEST/ncbi-igblast-1.22.0-x64-linux.tar.gz\n",
        "!pip install Bio\n",
        "!git clone https://github.com/Eitan177/cloneseer.git\n",
        "!pip install git+https://github.com/aerijman/igblast-parser.git\n",
        "!pip install fastcluster\n",
        "!export IGDATA=\"/content/ncbi-igblast-1.22.0/bin\"\n",
        "!tar -xzvf ncbi-igblast-1.22.0-x64-linux.tar.gz\n",
        "import sys\n",
        "import subprocess\n",
        "import pandas as pd\n",
        "from io import StringIO\n",
        "import warnings\n",
        "import numpy as np\n",
        "from Bio import SeqIO\n",
        "import os\n",
        "os.chdir('/content/ncbi-igblast-1.22.0')\n",
        "os.mkdir('database')\n",
        "os.mkdir('database_clean')\n",
        "os.chdir('/content/ncbi-igblast-1.22.0/database')\n",
        "!wget https://www.imgt.org/download/V-QUEST/IMGT_V-QUEST_reference_directory/Homo_sapiens/IG/IGHV.fasta\n",
        "!wget https://www.imgt.org/download/V-QUEST/IMGT_V-QUEST_reference_directory/Homo_sapiens/IG/IGHD.fasta\n",
        "!wget https://www.imgt.org/download/V-QUEST/IMGT_V-QUEST_reference_directory/Homo_sapiens/IG/IGHJ.fasta\n",
        "os.chdir('/content/ncbi-igblast-1.22.0')\n",
        "# cleaning up IMGT germline sequence files\n",
        "input_imgt_ref = 'database/IGHV.fasta'\n",
        "#input_imgt_ref =\"/content/ighv_functional.fasta\"\n",
        "output_imgt_ref = 'database_clean/IGHV_clean.fasta'\n",
        "\n",
        "cmd = ['bin/edit_imgt_file.pl', input_imgt_ref , '>', output_imgt_ref]\n",
        "\n",
        "\n",
        "fh = open(output_imgt_ref, 'w')\n",
        "a = subprocess.Popen(cmd, stdout=fh)\n",
        "a.wait()\n",
        "\n",
        "input_imgt_ref = 'database/IGHD.fasta'\n",
        "output_imgt_ref = 'database_clean/IGHD_clean.fasta'\n",
        "\n",
        "cmd = ['bin/edit_imgt_file.pl', input_imgt_ref , '>', output_imgt_ref]\n",
        "\n",
        "fh = open(output_imgt_ref, 'w')\n",
        "a = subprocess.Popen(cmd, stdout=fh)\n",
        "a.wait()\n",
        "input_imgt_ref = 'database/IGHJ.fasta'\n",
        "output_imgt_ref = 'database_clean/IGHJ_clean.fasta'\n",
        "\n",
        "cmd = ['bin/edit_imgt_file.pl', input_imgt_ref , '>', output_imgt_ref]\n",
        "\n",
        "# display result (and any error) in notebook. No file will be saved\n",
        "# subprocess.run(cmd.split(), capture_output=True)\n",
        "\n",
        "# save cmd output to file\n",
        "fh = open(output_imgt_ref, 'w')\n",
        "a = subprocess.Popen(cmd, stdout=fh)\n",
        "a.wait()\n",
        "cmd = ['bin/makeblastdb', '-parse_seqids', '-dbtype', 'nucl', '-in', '/content/ncbi-igblast-1.22.0/database_clean/IGHD_clean.fasta']\n",
        "\n",
        "a = subprocess.Popen(cmd, stdout=subprocess.PIPE, cwd='/content/ncbi-igblast-1.22.0')\n",
        "a.wait()\n",
        "cmd = ['bin/makeblastdb', '-parse_seqids', '-dbtype', 'nucl', '-in', '/content/ncbi-igblast-1.22.0/database_clean/IGHV_clean.fasta']\n",
        "\n",
        "a = subprocess.Popen(cmd, stdout=subprocess.PIPE, cwd='/content/ncbi-igblast-1.22.0')\n",
        "a.wait()\n",
        "cmd = ['bin/makeblastdb', '-parse_seqids', '-dbtype', 'nucl', '-in', '/content/ncbi-igblast-1.22.0/database_clean/IGHJ_clean.fasta']\n",
        "\n",
        "a = subprocess.Popen(cmd, stdout=subprocess.PIPE, cwd='/content/ncbi-igblast-1.22.0')\n",
        "a.wait()\n",
        "os.chdir('/content/ncbi-igblast-1.22.0')\n",
        "\n",
        "os.chdir('/content/')\n",
        "!mkdir /content/cloneseer_out/\n",
        "!unzip cloneseer/Input_Folder_Rep1.zip\n",
        "!unzip cloneseer/Input_Folder_Rep2.zip\n",
        "!pip install streamlit==1.23.0\n",
        "os.chdir('/content/ncbi-igblast-1.22.0')"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!python /content/cloneseer/lymphotrack_main_local.py --sample_sheet /content/cloneseer/R14-078_R14-079_samplesheet.xlsx --clone_length_fr1 40 --clone_length_fr2 30 --clone_length_fr3 20 --independent_kmer 1 --must_be_identical 0.95 --save_path /content/cloneseer_out --skiprows 0 --AnalyzePositiveControl 0 --Match_Clone_with_FR 0 --readrows 50 --Input_Nanograms 50 --aggregate 0"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zIgVl6ulhd1y",
        "outputId": "577bfd3c-2808-4340-ccf0-31d8438d5ca1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/usr/local/lib/python3.12/dist-packages/Bio/pairwise2.py:278: BiopythonDeprecationWarning: Bio.pairwise2 has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.PairwiseAligner as a replacement, and contact the Biopython developers if you still need the Bio.pairwise2 module.\n",
            "  warnings.warn(\n",
            "2025-09-05 20:28:27.737 \n",
            "  \u001b[33m\u001b[1mWarning:\u001b[0m to view this Streamlit app on a browser, run it with the following\n",
            "  command:\n",
            "\n",
            "    streamlit run /content/cloneseer/lymphotrack_main_local.py [ARGUMENTS]\n",
            "2025-09-05 20:28:27.737 `st.experimental_memo` is deprecated. Please use the new command `st.cache_data` instead, which has the same behavior. More information [in our docs](https://docs.streamlit.io/library/advanced-features/caching).\n",
            "2025-09-05 20:28:27.737 No runtime found, using MemoryCacheStorageManager\n",
            "2025-09-05 20:28:28.751 `st.experimental_memo` is deprecated. Please use the new command `st.cache_data` instead, which has the same behavior. More information [in our docs](https://docs.streamlit.io/library/advanced-features/caching).\n",
            "2025-09-05 20:28:28.751 No runtime found, using MemoryCacheStorageManager\n",
            "argument parsing\n",
            "0\n",
            "Sample ID Path      /content/Input_Folder_Rep2\n",
            "Sample ID                              R14-078\n",
            "Replicate Number                             2\n",
            "Clone File Path                      /content/\n",
            "Clone File                        R14-078.xlsx\n",
            "Descriptors                                NaN\n",
            "Name: 0, dtype: object\n",
            "about to process sample sheet\n",
            "made dictionary\n",
            "R14-078_FR1_Clone1\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n",
            "Framework 1\n",
            "Warning: Auxilary data file could not be found\n",
            "2025-09-05 20:28:32.986 No runtime found, using MemoryCacheStorageManager\n",
            "17\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1073', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1067', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1161', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1155', ...)}\n",
            "Warning: Auxilary data file could not be found\n",
            "N/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCACCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTTTGACCGCTGCGTACACTGCCGTGTATTACTTTTCTAGAGACTACAAGAACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGATCTCCCTGAACCTGAGCTCTGTGACCGCTGCGTACACTGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTAACCACTGCTTACACGTCCGTTTATTACTTTTCGAGAGAGTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGAGGACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGAAACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTGCGAGACACTACAATTACGACGGGATGGACGTCTGGGGCCAAN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTGCTAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTTCGAGAGACTACAACTACTACGGTATGGTCGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTTCTAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTTTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTTTTCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCTTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGTACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCTGACACGGCCGTGTATTACTGTTCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in CACTGTCTCTGGTGGCTCCGTCAGCAGTGGTAGATACTACTGGAGTTGGATCCGGCAGCCCCCAGGGAAGGGCCTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGAGAACACGGCAGAGTATTACTATGTGAGAGACTACACCTCCTACGGTATGGACGTCTGGGGCCAAGGGACCACERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text_font='monospace' [no close matches] {renderer: GlyphRenderer(id='p1241', ...)}\n",
            "/content/cloneseer/view_alignments_0401.py:287: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
            "  table_w_sequence_count_sorted_w_filtered_vdj = table_w_sequence_count_sorted_w_filtered_vdj[table_w_sequence_count_sorted_w_filtered_vdj[['mother_clone_distance','_merge']].apply(lambda x: x[0]<3 or x[1]=='both',axis=1)].drop(columns=[\"_merge\"])\n",
            "/content/cloneseer/view_alignments_0401.py:293: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
            "  table_w_sequence_count_sorted_w_filtered_vdj= table_w_sequence_count_sorted_w_filtered_vdj[table_w_sequence_count_sorted_w_filtered_vdj[['top_v_gene_match_x','top_v_gene_match_y','v_end_x','v_end_y','j_start_x','j_start_y','insert_trinucleotide','delete_trinucleotide','mother_clone_distance']].apply(lambda x: (re.sub(';.+','',str(x[0]))==re.sub(';.+','',str(x[1])) and str(x[2])==str(x[3]) and str(x[4])==str(x[5])) or (str(x[0])==str(x[1]) and ((x[6] % 3 == 0 and x[6]>2) or (x[7] % 3 == 0 and x[7]>2)) or (x[8]<3)),axis=1)]\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1326', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1320', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1414', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1408', ...)}\n",
            "R14-078_FR1_Clone1\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n",
            "Framework 2\n",
            "Warning: Auxilary data file could not be found\n",
            "3\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n",
            "[0, 5505, 5996]\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1496', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1502', ...)}\n",
            "[0, 2686, 3177]\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1590', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1584', ...)}\n",
            "Warning: Auxilary data file could not be found\n",
            "N/A not flanked by C and F in AGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in GGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAATTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAACCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text_font='monospace' [no close matches] {renderer: GlyphRenderer(id='p1670', ...)}\n",
            "/content/cloneseer/view_alignments_0401.py:287: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
            "  table_w_sequence_count_sorted_w_filtered_vdj = table_w_sequence_count_sorted_w_filtered_vdj[table_w_sequence_count_sorted_w_filtered_vdj[['mother_clone_distance','_merge']].apply(lambda x: x[0]<3 or x[1]=='both',axis=1)].drop(columns=[\"_merge\"])\n",
            "/content/cloneseer/view_alignments_0401.py:293: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
            "  table_w_sequence_count_sorted_w_filtered_vdj= table_w_sequence_count_sorted_w_filtered_vdj[table_w_sequence_count_sorted_w_filtered_vdj[['top_v_gene_match_x','top_v_gene_match_y','v_end_x','v_end_y','j_start_x','j_start_y','insert_trinucleotide','delete_trinucleotide','mother_clone_distance']].apply(lambda x: (re.sub(';.+','',str(x[0]))==re.sub(';.+','',str(x[1])) and str(x[2])==str(x[3]) and str(x[4])==str(x[5])) or (str(x[0])==str(x[1]) and ((x[6] % 3 == 0 and x[6]>2) or (x[7] % 3 == 0 and x[7]>2)) or (x[8]<3)),axis=1)]\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n",
            "[0, 5505, 5996]\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1755', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1749', ...)}\n",
            "[0, 2686, 3177]\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1837', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1843', ...)}\n",
            "R14-078_FR1_Clone1\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n",
            "fr is three and len clone is greater than 200\n",
            "Framework 3\n",
            "Warning: Auxilary data file could not be found\n",
            "34\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n",
            "[26397, 27070, 27874, 28006, 27978, 26980, 24877, 27039, 24555, 27766, 28142, 26315, 27852, 27011, 27959, 26952, 24126, 26900, 27790, 27990, 27909, 0, 27255, 26653, 27920, 27902, 25995, 27946, 26863, 28158, 27024, 27830, 25219]\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1925', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p1931', ...)}\n",
            "[3022, 3695, 4499, 4631, 4603, 3605, 1502, 3664, 1180, 4391, 4767, 2940, 4477, 3636, 4584, 3577, 751, 3525, 4415, 4615, 4534, 0, 3880, 3278, 4545, 4527, 2620, 4571, 3488, 4783, 3649, 4455, 1844]\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p2013', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p2019', ...)}\n",
            "Warning: Auxilary data file could not be found\n",
            "N/A not flanked by C and F in ACACGGCCATGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGCGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTATATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGCATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTACTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTAATGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCAAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAAAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAAACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGCATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTACGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACATCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGCCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGACCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGATCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGGCCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTTTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGATGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTACAACTACTATGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTACAGCTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGACTATAACTATTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGAGAGGCTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGGGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGGGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGATCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGCGTGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGATCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTGTGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTATTACTGTTCGAGAGACTACAACTACTACGGGATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCCGTGTGTTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCTGTGTATTACTGTGCGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACN/A not flanked by C and F in ACACGGCTGTGTATTACTGTGTGAGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCAN/A not flanked by C and F in ACACGGTCGTGTATTACTGTGCGGGAGACTACAACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text_font='monospace' [no close matches] {renderer: GlyphRenderer(id='p2099', ...)}\n",
            "/content/cloneseer/view_alignments_0401.py:287: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
            "  table_w_sequence_count_sorted_w_filtered_vdj = table_w_sequence_count_sorted_w_filtered_vdj[table_w_sequence_count_sorted_w_filtered_vdj[['mother_clone_distance','_merge']].apply(lambda x: x[0]<3 or x[1]=='both',axis=1)].drop(columns=[\"_merge\"])\n",
            "/content/cloneseer/view_alignments_0401.py:291: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
            "  table_w_sequence_count_sorted_w_filtered_vdj= table_w_sequence_count_sorted_w_filtered_vdj[table_w_sequence_count_sorted_w_filtered_vdj[['mother_clone_distance','insert_trinucleotide','delete_trinucleotide']].apply(lambda x: x[0]<10 and ((x[1] % 3 == 0 and x[1]<=15) and (x[2] % 3 == 0 and x[2]<=15)),axis=1)]\n",
            "/usr/local/lib/python3.12/dist-packages/Bio/SeqRecord.py:228: BiopythonDeprecationWarning: Using a string as the sequence is deprecated and will raise a TypeError in future. It has been converted to a Seq object.\n",
            "  warnings.warn(\n",
            "[26607, 27026, 27852, 27994, 28130, 26980, 25653, 27722, 24555, 27793, 28142, 26315, 27887, 27817, 27909, 26952, 24126, 26900, 27753, 27978, 27967, 0, 27211, 26397, 27928, 27880, 25995, 27954, 26863, 27011, 27830, 24877]\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p2184', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p2178', ...)}\n",
            "[3232, 3651, 4477, 4619, 4755, 3605, 2278, 4347, 1180, 4418, 4767, 2940, 4512, 4442, 4534, 3577, 751, 3525, 4378, 4603, 4592, 0, 3836, 3022, 4553, 4505, 2620, 4579, 3488, 3636, 4455, 1502]\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p2266', ...)}\n",
            "ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text='text' [no close matches] {renderer: GlyphRenderer(id='p2272', ...)}\n",
            "1\n",
            "Sample ID Path      /content/Input_Folder_Rep2\n",
            "Sample ID                              R14-079\n",
            "Replicate Number                             2\n",
            "Clone File Path                      /content/\n",
            "Clone File                        R14-078.xlsx\n",
            "Descriptors                                NaN\n",
            "Name: 1, dtype: object\n",
            "about to process sample sheet\n",
            "made dictionary\n",
            "R14-078_FR1_Clone1\n"
          ]
        }
      ]
    }
  ]
}